src/python/pants/backend/jvm/tasks/jvm_compile/zinc/zinc_compile.py

```diff
@@ -207,6 +209,20 @@ def __init__(self, *args, **kwargs):
     # Validate zinc options.
     ZincCompile.validate_arguments(self.context.log, self.get_options().whitelisted_args,
                                    self._args)
+    if self.execution_strategy == self.HERMETIC:
+      if fast_relpath(self.get_options().pants_workdir, get_buildroot()).startswith('..'):
```

Comment:
`fast_relpath` eagerly fails if `arg1 does not contain arg2`, so it will never result in a `..` prefix.